java/query/src/main/java/sleeper/query/model/QueryOrLeafQuery.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2022-2023 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package sleeper.query.model;
+
+import sleeper.configuration.properties.table.TableProperties;
+import sleeper.configuration.properties.table.TablePropertiesProvider;
+import sleeper.query.model.output.ResultsOutputInfo;
+import sleeper.query.tracker.QueryStatusReportListener;
+
+import java.util.Objects;
+
+public class QueryOrLeafQuery {
```

Comment:
I can see why this was called `QueryOrLeafQuery` but `QueryOrLeafPartitionQuery` is more accurate and not too long for a class name.